src/main/java/org/elastos/did/DIDStore.java

```diff
@@ -362,41 +394,58 @@ public String publishDid(DID did, int confirms,
 		if (doc.isDeactivated())
 			throw new DIDStoreException("DID already deactivated.");
 
+		String lastTxid = null;
 		DIDDocument resolvedDoc = did.resolve();
 		if (resolvedDoc != null) {
 			if (resolvedDoc.isDeactivated())
 				throw new DIDStoreException("DID already deactivated.");
 
-			String localTxid = doc.getTransactionId();
-			if (localTxid == null)
-				throw new DIDStoreException("DID document not up-to-date");
+			if (!force) {
+				String localTxid = doc.getMeta().getTransactionId();
+				String localSignature = doc.getMeta().getSignature();
+
+				String resolvedTxid = resolvedDoc.getTransactionId();
+				String reolvedSignautre = resolvedDoc.getProof().getSignature();
+
+				if (localTxid == null && localSignature == null)
+					throw new DIDStoreException("DID document not up-to-date");
+
+				if (localTxid != null && !localTxid.equals(resolvedTxid))
```

Comment:
I when the express of "localTxid != null && !localTxid.equals(resolvedTxid)" is true,  there is no need to check the express of "localSignature != null && !localSignature.euqals(resolvedSignature)".

Reply:
you can think this is a double check. nomally this expression should be false except the store's integrity was broken.